platform/wab/src/wab/client/components/sidebar-tabs/ProjectPanel/NavigationRows.tsx

```diff
@@ -64,29 +109,79 @@ export function NavigationHeaderRow({
       style={{ height: HEADER_HEIGHT }}
       className="bt-dim"
       showActions={true}
-      actions={RowAddAction({ onAdd, toggleExpand, isOpen: props.isOpen })}
+      actions={
+        <IconButton
+          onClick={async (e) => {
+            e.stopPropagation();
+            await onAdd();
+            if (!props.isOpen) {
+              toggleExpand();
+            }
+          }}
+        >
+          <Icon icon={PlusIcon} />
+        </IconButton>
+      }
       {...props}
     />
   );
 }
 
-interface NavigationFolderRowProps extends RowGroupProps, RowAddActionProps {
+interface NavigationFolderRowProps extends RowGroupProps {
+  folder: FolderElement;
+  matcher: Matcher;
   indentMultiplier: number;
+  toggleExpand: () => void;
 }
 
 export function NavigationFolderRow({
-  onAdd,
-  toggleExpand,
+  folder,
+  matcher,
   indentMultiplier,
-  ...props
+  isOpen,
+  toggleExpand,
 }: NavigationFolderRowProps) {
+  const [renaming, setRenaming] = React.useState(false);
   return (
     <RowGroup
       style={{ height: ROW_HEIGHT, paddingLeft: indentMultiplier * 16 + 12 }}
       showActions={true}
-      actions={RowAddAction({ onAdd, toggleExpand, isOpen: props.isOpen })}
-      {...props}
-    />
+      menu={
+        <FolderContextMenu
+          onAdd={async () => {
+            if (!isOpen) {
+              toggleExpand();
+            }
+            await folder.onAdd();
+          }}
+          arenaDisplay={getArenaDisplay(folder.sectionType)}
+          onSelectRename={() => setRenaming(true)}
+          onDelete={() => folder.onDelete(folder)}
+        />
+      }
+      actions={<div></div>}
+      groupSize={folder.count}
+      isOpen={isOpen}
+    >
+      <EditableLabel
```

Comment:
I copied this usage direction from NavigationArenaRow, and fixed up the UI a bit in a subsequent commit. I wasn't sure if `programmaticallyTriggered` is still necessary, I didn't see the behavior mentioned in the comment but may have missed something.